pinot-core/src/main/java/org/apache/pinot/core/segment/index/loader/defaultcolumn/V3DefaultColumnHandler.java

```diff
@@ -61,21 +63,27 @@ protected void updateDefaultColumn(String column, DefaultColumnAction action)
           "Default value indices for column: " + column + " cannot be removed for V3 format segment.");
     }
 
-    // Create new dictionary and forward index, and update column metadata
-    createColumnV1Indices(column);
```

Comment:
Hmm.. Does this mean that we always created dictionary even if the column had raw index? Of course, it should not matter, since the column has only one default value, and having the dictonary is better (storage-wise). As long as segment metadata is treated as the source of truth, we are good.

Reply:
This is existing behavior (I mean not changed as part of this PR. Github doesn't do a good job of recognizing even little bit of code movement ).

I noticed this as well. I am not sure why we do this. Can look into it separately